include/LightGBM/bin.h

```diff
@@ -56,6 +59,28 @@ inline static void HistogramSumReducer(const char* src, char* dst, int type_size
   }
 }
 
+inline static void Int32HistogramSumReducer(const char* src, char* dst, int type_size, comm_size_t len) {
+  const int64_t* src_ptr = reinterpret_cast<const int64_t*>(src);
+  int64_t* dst_ptr = reinterpret_cast<int64_t*>(dst);
```

Comment:
is it okay to use int64_t pointer?

Reply:
I think the alignments of these buffer vectors is 32 bytes now. So, it should be safe to do this.
https://github.com/microsoft/LightGBM/blob/788e1aa6ae08a14b6bf6f2799b5c4d61426b2079/src/treelearner/parallel_tree_learner.h#L89-L91